ion-schema/src/isl/isl_constraint.rs

```diff
@@ -433,12 +439,37 @@ pub mod v_2_0 {
         )
     }
 
-    /// Creates an `annotations` constraint using [str]s and [Element]s specified inside it
-    pub fn annotations<'a, A: IntoIterator<Item = &'a str>, B: IntoIterator<Item = Element>>(
-        annotations_modifiers: A,
-        annotations: B,
+    /// Creates an `annotations` constraint using a list of valid annotations and specify whether annotations are required or closed or both.
+    pub fn simple_annotations<A: IntoIterator<Item = Element>>(
+        is_required: bool,
+        is_closed: bool,
```

Comment:
question (non-blocking)—Is it possible to model this a different way so that it won't compile unless one of these is true?

Reply:
How about adding a note in the doc comment that by default required would be set to `true` if both of them are passed as `false`? or set function argument to be an enum representing `Required`, `Closed` and `RequiredAndClosed`?

Reply:
> How about adding a note in the doc comment that by default required would be set to true if both of them are passed as false? 

This is not a great idea. It's better to return an `Err` (or even panic) than it is to change an explicitly provided argument.

> or set function argument to be an enum representing Required, Closed and RequiredAndClosed?

Something like this could work. Another option might be to have overloaded functions for `annotations_closed`, `annotations_closed_required`, and `annotations_required`.

I think for consumers of the library, an enum is probably the best choice. For internal use (i.e. for reading Ion Schema), it might be more useful to keep this signature but return `Err` if there's a problem.

Reply:
I just checked that for `valid_values`  we return an error when there are annotations other than `range`. I think probably something similar would make sense for here.